backend/pkg/helm/repository.go

```diff
@@ -0,0 +1,261 @@
+package helm
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	zlog "github.com/rs/zerolog/log"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	defaultNewConfigFileMode   os.FileMode = os.FileMode(0644)
+	defaultNewConfigFolderMode os.FileMode = os.FileMode(0770)
+)
+
+// add repository
+
+type AddUpdateRepoRequest struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+	// TODO: Figure out how to support auth
```

Comment:
There's a TODO here.

Reply:
This is fine to improve later IMO.

Reply:
closing duplicate comment in favor of https://github.com/headlamp-k8s/headlamp/pull/1079#discussion_r1173458174